File: 18/36.cpp
#include <iostream>
#include <fstream>

using namespace std;

// 345 (Даже написал на плюсах, не знаю, что с ней не так)

int main()
{
  ifstream inp("dat/18-17.txt");

  double p = INT8_MAX;
  double msum = INT8_MIN;
  double sum = 0;

  while (!inp.eof())
  {
    double n;
    inp >> n;

    if (n < p)
    {
      sum += n;
      if (sum > msum)
        msum = sum;
    }
    else
      sum = 0;

    p = n;
  }

  cout << (int)msum << endl;

  return 0;
}

File: 18/36.py
# 495 (я дурашка)

with open('./dat/18-17.txt', 'r') as f:
  p = float('+inf')
  s = 0
  ms = float('-inf')

  for sn in f.readlines():
    n = float(sn)
    if n < p:
      s += n
      if s > ms:
        ms = s
    else:
      s = n
    p = n

print(int(ms))
